src/orange/detail/BIHBuilder.cc

```diff
@@ -0,0 +1,222 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/detail/BIHBuilder.cc
+//---------------------------------------------------------------------------//
+#include "BIHBuilder.hh"
+
+#include "BoundingBoxUtils.hh"
+
+namespace celeritas
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Variadic-templated struct for overloaded operator() calls.
+ */
+template<typename... Ts>
+struct Overload : Ts...
+{
+    using Ts::operator()...;
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * "Deduction guide" for instantiating Overload objects w/o specifying types.
+ */
+template<class... Ts>
+Overload(Ts...) -> Overload<Ts...>;
```

Comment:
```suggestion
Overload(Ts&&...) -> Overload<Ts...>;
```
to get perfect forwarding

Reply:
Fixed in OrangeParams.cc as well